src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -66,32 +193,50 @@ export default () => {
             // applied in their correct and specific order. If this queue was not paused, then there would be a lot of
             // onyx data being applied while we are fetching the missing updates and that would put them all out of order.
             SequentialQueue.pause();
-            let canUnpauseQueuePromise;
+            let queryPromise: Promise<Response | void>;
 
             // The flow below is setting the promise to a reconnect app to address flow (1) explained above.
             if (!lastUpdateIDAppliedToClient) {
                 Log.info('Client has not gotten reliable updates before so reconnecting the app to start the process');
 
+                deferredUpdateBeforeReconnect = updateParams;
+
                 // Since this is a full reconnectApp, we'll not apply the updates we received - those will come in the reconnect app request.
-                canUnpauseQueuePromise = App.finalReconnectAppAfterActivatingReliableUpdates();
+                queryPromise = App.finalReconnectAppAfterActivatingReliableUpdates().then(() => {
+                    // If "updateAfterReconnectApp" is set, it means that case (1) was triggered and we need to apply the update that was queued before the reconnectApp
+                    if (!deferredUpdateBeforeReconnect) {
+                        return;
+                    }
+
+                    OnyxUpdates.apply(deferredUpdateBeforeReconnect);
+                    deferredUpdateBeforeReconnect = undefined;
+                });
```

Comment:
I've been thinking about this one specifically... I don't think we need to apply the deferred updates here. If you do a finalReconnect, you'll receive everything, including the update we've been storing, which is different than fetching the missing onyxUpdates that will only return the ones you're missing.

That being the case, let's drop the `deferredUpdateBeforeReconnect` property and remove this code after the finalReconnectApp.